pygmt/figure.py

```diff
@@ -394,3 +433,33 @@ def _repr_html_(self):
         subplot,
         text,
     )
+
+
+def set_display(method=None, dpi=None, width=None):
```

Comment:
```suggestion
def set_display(method=None):
```

Personally I don't see a use-case for setting a global `dpi` or `width` (but happy to be proven wrong). My suggestion is that we only let this `set_display` function set the method and not do anything about the dpi or width. Haven't looked too closely, but I think most of the changes were made in ca0ecb3d649bbf00b380e8848b3c82b9cd28a68c and could be partially reverted there. Not doing anything about the dpi/width would also greatly reduce the diff on this PR and make it easier to review. Discuss first though.

Reply:
We can discuss the global *dpi* and *width* setting in a separate PR.